### PR TITLE
Work branch

### PR DESCRIPTION
**Pull request by Josphat205**
- Created a new Rails API app called 'hello-rails-back-end'.
- Initialized your project with Git.
- make sure that your project has a Postgres database set up.
- Created a table for storing your messages and create 5 different greetings.
- Created an API endpoint that selects a random greeting from your table (you will need a controller with an action and rails route).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,148 @@
...
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+_My acknoledgemet goes to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) - and the author[ author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=). You can also acknoledge the designer [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) if you wish._
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](LICENSE.txt) licensed.
```
Comment (Ismailco) [inline README.md:146]: - [ ] Kindly make sure all the links in your README file are working properly, the LICENSE does not exist in the repo, kindly consider adding a LICENSE and edit the name from `LICENSE.txt` to `LICENSE` so GitHub can automatically detect it and associate it with this repo

File: app/models/greeting.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Greeting < ApplicationRecord
+  # frozen_string_literal: true
```
Comment (Ismailco) [inline app/models/greeting.rb]: - [ ] Kindly make sure to remove old commented code from your repo so it appear clean and easy to navigate

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Josphat205, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
- [x] Kindly make sure all the links in your README file are working properly, the LICENSE does not exist in the repo, kindly consider adding a LICENSE and edit the name from `LICENSE.txt` to `LICENSE` so GitHub can automatically detect it and associate it with this repo
- [x] The Front end produces an error because you're trying to render the greeting message before it's fetched from the API, which produces an error of `message` undefined. I highly suggest moving the logic (fetch message) from the `Home` component into the Reducer (redux) or adding a condition before rendering the message like so: 

```javascript
 {greeting.message && (
        <p>
          {greeting.message.message} posted on {greeting.message.created_at}
        </p>
      )}
```  
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/greeting.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
 class Greeting < ApplicationRecord
-  # frozen_string_literal: true
+  validates :name, presence: true
```
Comment (Kamba56) [inline app/models/greeting.rb]: - [ ] There is no name field in the greetings table. I notice you are validating for the presence of a name while there is no name in the greetings table. Kindly remove this please as it is returning an error.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @Josphat205, 

Great work on making the changes requested by a previous reviewer 👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [x] Great job consuming the greetings API!👏👏👏
- [x] !👏👏👏

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @Josphat205 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Implemented all the changes from the previous reviewer :white_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**